### PR TITLE
Feature/#82 stacked bar chart rendering instead of grouped bar chart

### PR DESCRIPTION
**Pull request by rousan**

### CODE REVIEW

File: packages/visual-layer/src/helpers/index.js [hunk 1]
```diff
@@ -143,14 +143,36 @@ export const getEncodingFieldInf = (encoding, fieldsConfig) => {
...
 
+/**
+ * Retrieves the nearest groupBy aggFn function in its derivation cycle for the specified field.
+ *
+ * @param {DataModel} dataModel - The target DataModel instance.
+ * @param {string} fieldName - The target field name.
+ * @return {string} Returns the aggFn name.
+ */
+export const retrieveGroupByAggFn = (dataModel, fieldName) => {
+    let next = dataModel;
```
Comment (ranajitbanerjee) [inline packages/visual-layer/src/helpers/index.js]: Keep this method in utils

File: packages/muze-utils/src/common-utils.js [hunk 2]
```diff
@@ -1467,6 +1467,56 @@ const nextAnimFrame = window.requestAnimationFrame || window.webkitRequestAnimat
...
+                    return aggFn;
+                }
+            }
+        }
+    } while (next = next.getParent());
+
+    return null;
+};
+
+const retrieveNearestGroupByReducers = (dataModel) => {
```
Comment (ranajitbanerjee) [inline packages/muze-utils/src/common-utils.js]: `retrieveNearestGroupByReducers` and `retrieveGroupByAggFn` almost have same code. Can't we make it a single method? If no `fieldName` is passed, then the method will return the aggregations functions of all measures

Comment (rousan) [reply packages/muze-utils/src/common-utils.js]: yeah, then we can implement a method as `retrieveNearestGroupByReducers(datamodel, ...fieldNames)`

File: packages/visual-layer/src/helpers/index.js [hunk 3]
```diff
@@ -162,11 +164,21 @@ export const getValidTransform = (layerConfig, fieldsConfig, encodingFieldInf) =
...
     } = encodingFieldInf;
     const groupByField = transform.groupBy;
+    const fieldsConfig = dataModel.getFieldsConfig();
     const groupByFieldMeasure = fieldsConfig[groupByField] && fieldsConfig[groupByField].def.type === FieldType.MEASURE;
+    const isCustomTransformTypeProvided = !!getObjProp(context._customConfig, 'transform', 'type');
     transformType = transform.type;
     if (!xField || !yField || groupByFieldMeasure || !groupByField || xFieldType === FieldType.DIMENSION &&
         yFieldType === FieldType.DIMENSION) {
         transformType = IDENTITY;
+    } else if (!isCustomTransformTypeProvided && groupByField && xFieldType !== yFieldType) {
```
[the remark this answers is not in the document]
Comment (adotg) [reply packages/visual-layer/src/helpers/index.js]: @ranajitbanerjee @rousan

[the remark this answers is not in the document]
Comment (ranajitbanerjee) [reply packages/visual-layer/src/helpers/index.js]: Yes

File: packages/muze-utils/src/common-utils.js [hunk 4]
```diff
@@ -1467,6 +1468,36 @@ const nextAnimFrame = window.requestAnimationFrame || window.webkitRequestAnimat
...
     };
 
+const retrieveNearestGroupByReducers = (dataModel, ...measureFieldNames) => {
+    let nearestReducers = {};
+    let next = dataModel;
+    do {
+        const derivations = next.getDerivations();
+        if (derivations && derivations.length >= 1) {
+            const latestDerivation = derivations[derivations.length - 1];
+            if (latestDerivation.op === DM_OPERATION_GROUP) {
```
Comment (ranajitbanerjee) [inline packages/muze-utils/src/common-utils.js]: This fails in case of compose operation.

**Review by ranajitbanerjee — APPROVED** [review]

**Review by adotg — CHANGES_REQUESTED** [review]

**Pull request merged**